proto/spire-next/types/status.proto

```diff
@@ -9,3 +9,19 @@ message Status {
     // A developer-facing error message.
     string message = 2;
 }
+
+message PermissionDeniedDetails {
+    enum Reason {
+        // Reason unknown.
+        UNKNOWN = 0;
+        // Agent identity has expired.
+        AGENT_EXPIRED = 1;
+        // Identity is not an attested agent.
+        AGENT_NOT_ATTESTED = 2;
+        // Identity is not the active agent identity.
+        AGENT_NOT_ACTIVE = 3;
```

Comment:
I wonder if `AGENT_SERIAL_MISMATCH` may be more self-describing than `AGENT_NOT_ACTIVE`, wdyt?

Reply:
Hmm, maybe? The serial number is an implementation detail on how the server determines that the agent is using a stale SVID but the high level state is that the agent used in inactive SVID.

Reply:
Choosing `AGENT_SERIAL_MISMATCH` is more tied to the current implementation. IMHO I would keep `AGENT_NOT_ACTIVE` which hides the underlying logic and gives us more flexibility for changes in the future.

Reply:
+1 works for me